modules/delegates/api/fetchDelegates.ts

```diff
@@ -456,10 +456,20 @@ export async function fetchDelegatesPaginated({
   const [githubDelegates, allDelegatesWithNamesAndLinks] = await fetchAndMergeDelegates(network);
 
   const constitutionalDelegatesAddresses = filterDelegateAddresses(allDelegatesWithNamesAndLinks, null, null);
-  const filteredDelegates = filterDelegateAddresses(allDelegatesWithNamesAndLinks, cvcs, searchTerm);
-  const { constitutionalDelegatesCount, shadowDelegatesCount, totalDelegatesCount } = getDelegatesCounts(
-    allDelegatesWithNamesAndLinks
+  const filteredDelegateAddresses = filterDelegateAddresses(
+    allDelegatesWithNamesAndLinks,
+    cvcs,
+    searchTerm,
+    delegateType
   );
+  const filteredDelegateEntries =
+    !searchTerm && !cvcs
```

Comment:
why dont you just refactor the filterDelegateAddresses to also return the data you need in fiteredDelegateEntries,

meaning, you change filterDelegateAddresses to a filterDelegates function, and then use the result